backend/common.go

```diff
@@ -165,6 +169,9 @@ type PluginContext struct {
 
 	// GrafanaConfig is the configuration settings provided by Grafana.
 	GrafanaConfig *GrafanaCfg
+
+	// UserAgent is the user agent of the Grafana client that initiated the gRPC request.
+	UserAgent *useragent.UserAgent
```

Comment:
Nit: Since `useragent.UserAgent` is just 3 strings and unlikely to change, maybe it make sense to have this as a value rather than as a pointer to avoid allocating on the heap and dealing with `nil`s? But not sure

Reply:
Yeah had originally had this, but then moved to keep things consistent with the existing fields we have (particularly `User` since it also had just string fields and can be nil if is an alerting request). 

It comes down to whether we want the field to potentially be nil / plugin devs to handle nil cases (all requests prior to Grafana 10.2.0, which would be the likely targeted version for releasing all of this) and we defend against panics as part of the API. Otherwise, plugin devs deal with a non pointer and may expect it to always be populated. WDYT?

Reply:
Also @andresmgot you might have an opinion on this - When no user agent sent from Grafana:
- Have user agent be nil (just like `User` is when coming from an alert query)
- Have user agent as empty struct (when no user agent received) where grafana version will be `""`

Reply:
Okay, makes sense since we have the same for `User`. Feel free to mark this as resolved if Andres agrees as well :smile:

Reply:
my +1 to be potentially nil, it's easier to check and easier to notice if a check is missing (with a painful `panic` :sweat_smile: )

Reply:
So it sounds like `nil` without any safeguards? 😄 In that case I will need to change it to behave the same as `User`.